wgpu-core/src/device/queue.rs

```diff
@@ -1209,6 +1209,13 @@ impl Queue {
                             }
                         }
 
+                        if first_error.is_some() {
+                            if let Ok(cmd_buf_data) = cmd_buf_data {
+                                cmd_buf_data.destroy(&command_buffer.device);
```

Comment:
thought: I guess that this is technically more correct than before. The diagnostics seem like they would be wrong if a user attempts to use these command buffers again, though (i.e., "You can't use this, this is destroyed." as opposed to "this is invalid").

Not concerned either way, though.